project/account/models.py

```diff
@@ -0,0 +1,101 @@
+from django.db import models
+from django.utils import timezone
+from django.contrib.auth.models import AbstractBaseUser, BaseUserManager
+
+
+class UserManager(BaseUserManager):
+    def create_user(self, email, username, first_name, last_name, password=None):
+        """
+        Creates and saves a user with the given email, username and password.
+        """
+        if not email:
+            raise ValueError('Users must have an email address')
+        if not username:
+            raise ValueError("Users must have a username")
+        if not first_name:
+            raise ValueError("Users must have a first name")
+        if not last_name:
+            raise ValueError("Users must have a last name")
+        if not password:
+            raise ValueError('Users must have a password')
+
+        user = self.model(
+            email=self.normalize_email(email),
+            username=username,
+            first_name=first_name,
+            last_name=last_name,
+        )
+
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_staffuser(self, email, username, first_name, last_name, password):
+        """
+        Creates and saves a staff user with the given email, username and password.
+        """
+        user = self.create_user(
+            email=self.normalize_email(email),
+            password=password,
+            username=username,
+            first_name=first_name,
+            last_name=last_name,
+        )
+        user.is_staff = True
+        user.save(using=self._db)
+        return user
+
+    def create_superuser(self, email, username, first_name, last_name, password):
+        """
+        Creates and saves a superuser with the given email, username and password.
+        """
+        user = self.create_user(
+            email=self.normalize_email(email),
+            password=password,
+            username=username,
+            first_name=first_name,
+            last_name=last_name,
+        )
+        user.is_staff = True
+        user.is_admin = True
```

Comment:
A staff is an admin an admin is a staff too. So, you should include an "is_superuser" field and make it 'true' for "superusers", 'false' for others.

Reply:
Well, there can be staff that isn't admin.